demo/src/main/java/com/novoda/storagepathfinder/demo/StoragePathsAdapter.kt

```diff
@@ -0,0 +1,29 @@
+package com.novoda.storagepathfinder.demo
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.novoda.storagepathfinder.StoragePath
+
+typealias Listener<T> = (T) -> Unit
+
+internal class StoragePathsAdapter(private val layoutInflater: LayoutInflater,
+                                   private val deviceStoragePaths: List<StoragePath>,
+                                   private val onAddFileClicked: Listener<StoragePath>) : RecyclerView.Adapter<StoragePathViewHolder>() {
+
+    init {
+        setHasStableIds(true)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StoragePathViewHolder =
+            StoragePathViewHolder(layoutInflater.inflate(R.layout.item_device_storage_root, parent, false))
+
+    override fun onBindViewHolder(holder: StoragePathViewHolder, position: Int) {
+        holder.bind(deviceStoragePaths[position], onAddFileClicked)
```

Comment:
Because the result of both `onBindViewHolder` and `Bind` is `Unit` this can also be defined with an expression body instead of a block body.

Reply:
The difference is quite minimal:
```
override fun onBindViewHolder(holder: StoragePathViewHolder, position: Int) { 
  holder.bind(deviceStoragePaths[position], onAddFileClicked)
}
```
vs

```
override fun onBindViewHolder(holder: StoragePathViewHolder, position: Int) = 
  holder.bind(deviceStoragePaths[position], onAddFileClicked)
```
As long as it is in a separate line (for long expressions), it saves an extra line, we are on a diet :trollface: